sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TableOutputResolver.scala

```diff
@@ -241,7 +241,7 @@ object TableOutputResolver {
         val extraCols = inputCols.filterNot(col => matchedCols.contains(col.name))
           .map(col => s"${toSQLId(col.name)}").mkString(", ")
         throw QueryCompilationErrors.incompatibleDataToTableExtraStructFieldsError(
```

Comment:
I think we should throw `QueryCompilationErrors.cannotWriteTooManyColumnsToTableError` or `cannotWriteNotEnoughColumnsToTableError` here if `colPath` is empty.